zap/src/output/luau/client.rs

```diff
@@ -517,6 +636,85 @@ impl<'src> ClientOutput<'src> {
 		}
 	}
 
+	fn push_return_functions(&mut self) {
+		let call = self.config.casing.with("Call", "call", "call");
+		let value = self.config.casing.with("Value", "value", "value");
+
+		for (i, fndecl) in self.config.fndecls.iter().enumerate() {
+			let id = i + 1;
+
+			self.push_line(&format!("{name} = {{", name = fndecl.name));
+			self.indent();
+
+			self.push_indent();
+			self.push(&format!("{call} = function("));
+
+			if let Some(ty) = &fndecl.args {
+				self.push(&format!("{value}: "));
+				self.push_ty(ty);
+			}
+
+			self.push(")\n");
+			self.indent();
+
+			self.push_write_event_id(id);
+
+			self.push_line("function_call_id += 1");
+
+			self.push_line("function_call_id %= 256");
+
+			self.push_line(&format!("if event_queue[{id}][function_call_id] then"));
+			self.indent();
+
+			self.push_line("function_call_id -= 1");
+			self.push_line("error(\"Zap has more than 256 events awaiting a response, and therefore this packet has been dropped.\")");
+
+			self.dedent();
+			self.push_line("end");
+
+			self.push_line("alloc(1)");
+			self.push_line("buffer.writeu8(outgoing_buff, outgoing_apos, function_call_id)");
+
+			if let Some(data) = &fndecl.args {
+				self.push_stmts(&ser::gen(data, value, self.config.write_checks));
+			}
+
+			match self.config.yield_type {
+				YieldType::Yield => {
+					self.push_line(&format!("event_queue[{id}][function_call_id] = coroutine.running()"));
+					self.push_line("local value = coroutine.yield()");
```

Comment:
we want to `return coroutine.yield()`, not just call it

Reply:
Fixed the bug that Promise/Future didn't assign `value`. Below the match it returns `value`.